CHANGELOG.md

```diff
@@ -46,6 +46,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   e.g. `color.error = { bg = "red", bold = true, underline = true }` in your
   `~/.jjconfig.toml`.
 
+* `branches([needle])` revset function now takes `needle` as an optional
+  argument and matches just the branches whose name contains `needle` (this is
+  backwards compatible as `branches()` is equivalent to `branches("")` and
+  continues to resolve to all local branch targets).
+
+* `remote_branches([branch_needle[, remote_needle]])` now takes `branch_needle`
+  and `remote_needle` as optional arguments and matches just the branches whose
+  name contains `branch_needle` and remote contains `remote_needle` (this also
+  is backwards compatible as `remote_branches()` is equivalent to
+  `remote_branches("")` or `remote_branches("", "")` and continues to resolve
+  to all remote branch targets).
+
```

Comment:
Nit: I feel these changelog entries are a bit too detailed. Maybe we can just tell that these functions now take optional parameters to filter things? Some thing like:

```
* `branches([needle])` revset function now takes an optional
  argument to filter branches by name.
```

Reply:
I don't know about this, personally I would want to know about the backwards compatibility when reading changelog of some existing feature (and anyways, these could be edited later when doing the next release), but I'm going to leave this comment open if you or anyone else has any thoughts.

Reply:
I agree that it's important to mention backwards compatibility, but I think it's mostly useful when backwards compatibility is *not* preserved. I'd remove at least the "this is backwards compatible as " part since that's should be the reader's default assumption.

Reply:
Makes sense, done.